xaitk_saliency/utils/gen_coco_sal.py

```diff
@@ -1,285 +1,74 @@
+                if 'score' in ann:
+                    score_array = np.zeros(num_classes)
+                    score_array[ann['category_id'] - min_cid] = ann['score']
+                elif 'prob' in ann:
+                    score_array = ann['prob']
+                # annotation has no scores
                 else:
-                    raise MismatchedLabelsError("Provided detections and detector have mismatched class labels.")
-            except ValueError:  # cast to int failed
-                # try and match key with category name
-                cat_id = _name_to_cat_id(key, coco_cats)
-                if(cat_id == -1):
-                    raise MismatchedLabelsError("Provided detections and detector have mismatched class labels.")
-                cat_map[key] = cat_id
-
-        return cat_map
-
-    def _dets_to_formatted_mat(
-        dets: Iterable[Iterable[Tuple[AxisAlignedBoundingBox, Dict[Hashable, float]]]],
-        cats: dict
-    ) -> np.ndarray:
-        """
-        Converts detections, as returned by an implementation of
-        ``DetectImageObjects``, into a detection matrix formatted for use with
-        an implementation of ``GenerateDetectorProposalSaliency``. This
-        involves matching the labels in the detections with categories from a
-        COCO style dataset.
-
-        :param dets: Detections, as returned by an implementation of
-            ``DetectImageObjects``.
-        :param cats: Dict of categories, as taken from the "categories" section
-            of a COCO style annotation dictionary.
-        :returns: Matrix of detections with shape
-            [nImgs x nDets x (4+1+nClasses)]. If the number of detections for
-            each image is not consistent, the matrix will be padded with rows
-            of ones, except for the objectness which is set to zero.
-        """
-
-        max_cat_id = max(cats.keys())
-
-        dets_mat_list = []
-        for img_dets in dets:
-
-            img_dets = list(img_dets)
-
-            if len(img_dets) == 0:
-                dets_mat_list.append(np.empty([0, 4+1+max_cat_id+1]))
-                continue
-
-            bbox_mat = np.asarray([
-                [
-                    bbox.min_vertex[0],
-                    bbox.min_vertex[1],
-                    bbox.max_vertex[0],
-                    bbox.max_vertex[1]
-                ] for bbox in [det[0] for det in img_dets]
-            ])
-
-            score_dicts = [det[1] for det in img_dets]
-
-            cat_map = _map_score_dict_keys(score_dicts[0].keys(), cats)
-
-            # assume class ids are arranged between 0 and the maximum id
-            scores_mat = np.empty([0, max_cat_id+1])
-            objectness = np.ones(len(img_dets))
-
-            for det_i, score_dict in enumerate(score_dicts):
-                score_array = np.zeros(max_cat_id+1)
+                    # do not include any non-prediction annotations
+                    continue
```

Comment:
Due to the same "continue isn't covered because CPython", we could also add the `# pragma: no cover` comment here.